### DISCUSSION TITLE
MVVM模式中调用控件方法的问题

### DISCUSSION THREAD

**Discussion by tsctsc6** [BYJRK/DotNet-Discussions#330] [MVVM]
在开发中，有时会遇到需要使用某个控件的某个方法的情况。比如某个图表，需要刷新一下，就要调用 `plot.Refresh()` 。
而MVVM中，业务逻辑代码是在ViewModel中。当然我可以把整个 `plot` 传入ViewModel中，但是我感觉这不符合MVVM的设计规范（前后端分离）。
那么要怎么做才能符合MVVM的设计规范呢？是不是要在ViewModel中包含 `Action<>` `Func<>` 之类的包装一下 `plot.Refresh()` ？

**Comment by BYJRK** [comment]
你可以给vm写一个委托，然后在view中注册，这样vm只要调用委托即可；
你还可以使用诸如Messenger的方式，vm发送消息，view去接收。

**Reply by tsctsc6** [reply]
好的，谢谢。
